### PR TITLE
Tests for escaping in scripts. Also note that only enum and record types can be typedef-ed

### LINKED ISSUE #13: Document schemadef limitations

### PR DESCRIPTION
**Pull request by kaushik-work**
Resolves #13

### CODE REVIEW

**Comment by kaushik-work** [comment]
common-workflow-language/cwltool#1271

**Comment by tetron** [comment]
This shows up a bug in cwltool that needs to be fixed.

File: concepts.md [hunk 1]
```diff
@@ -484,21 +483,44 @@ If the value of a field has no leading or trailing non-whitespace
...
+programming language, the use of `$(...)` (and `${...}` in the case of
+expressions) may conflict with the syntax of that language.  For
+example, when writing shell scripts, `$(...)` is used to execute a
+command in a subshell and replace a portion of the command line with
+the standard output of that command.
+
+The following escaping rules apply.  The scanner makes a single pass
+from start to end with 3-character lookahead.  After performing a
+replacement scanning resumes at the next character following the
+replaced substring.
```
Comment (mr-c) [inline concepts.md:515]: Needs to be added to the changelog

File: concepts.md [hunk 2]
```diff
@@ -499,6 +500,25 @@ must be treated as a string interpolation.  After interpolating the first
...
+
+The following quoting rules apply.  The scanner makes a single pass
+from start to end with 3-character lookahead.  After performing a
+replacement scanning resumes at the next character following the
+replaced substring.
+
+  1. The substrings `\$(` and `\${` are replaced by `$(` and `${`
+     respectively.  No parameter or expression evaluation
+     interpolation occurs.
+  2. A double backslash `\\` is replaced by a single backslash `\`.
```
Comment (kaushik-work) [inline concepts.md:520]: @tetron does this mean that https://github.com/common-workflow-language/cwltool/issues/1271 will remain?

**Comment by tetron** [comment]
It means that should be fixed.  Take a look at the new tests.

**Pull request closed without merging**